src/Vehicle/Vehicle.cc

```diff
@@ -2511,20 +2592,45 @@ void Vehicle::setCurrentMissionSequence(int seq)
 
 void Vehicle::sendMavCommand(int component, MAV_CMD command, bool showError, float param1, float param2, float param3, float param4, float param5, float param6, float param7)
 {
-    MavCommandQueueEntry_t entry;
+    bool updated = false;
+
+    if (_mavCommandQueue.size() > 1) {
```

Comment:
@dogmaphobic please review this part carefully. To prevent excessive command queuing any additional duplicate commands are collapsed into what's already queued (but not in flight). You only care about the last command getting out and acknowledged.